src/datasource.ts

```diff
@@ -0,0 +1,119 @@
+import {
+    DataQueryRequest,
+    DataQueryResponse,
+    DataSourceApi,
+    DataSourceInstanceSettings,
+    MutableDataFrame,
+    FieldType,
+} from '@grafana/data';
+
+import { MyQuery, MyDataSourceOptions, ListIncidentsResponse, Incident } from './types';
+
+import {
+    FetchResponse,
+    getBackendSrv,
+} from "@grafana/runtime";
+
+import { lastValueFrom } from 'rxjs';
+
+export class DataSource extends DataSourceApi<MyQuery, MyDataSourceOptions> {
+    url?: string;
+
+    constructor(instanceSettings: DataSourceInstanceSettings<MyDataSourceOptions>) {
+        super(instanceSettings);
+        this.url = instanceSettings.url;
+        this.annotations = {};
+    }
+
+    async query(options: DataQueryRequest<MyQuery>): Promise<DataQueryResponse> {
+        const { range } = options;
+        const from = range!.from.toISOString();
+        const to = range!.to.toISOString();
+
+        const promises = options.targets.map(async (target) => {
+            const response = this.doRequest(target, from, to);
```

Comment:
Isn't there a need for pagination?

Reply:
There is but I initially planned to add it as a next iteration of improvement, if I can do it today I'll do so otherwise file a GitHub issue